destination/destination_integration_test.go

```diff
@@ -346,4 +350,61 @@ func TestWrite_CreateStreamIfNotExists(t *testing.T) {
 		assertWrittenRecordsOnStream(ctx, is, streamName2, recs2)
 		assertWrittenRecordsOnStream(ctx, is, streamName3, recs3)
 	})
+
+	t.Run("errors if option set to false in single collection mode", func(t *testing.T) {
+		logger := zerolog.New(zerolog.NewTestWriter(t))
+		ctx := logger.WithContext(context.Background())
+		is := is.New(t)
+		con := newDestination()
+
+		streamName := testutils.RandomStreamName("create_stream")
+
+		config := testutils.GetTestConfig(streamName)
+		config["createIfNotExists"] = "false"
+		err := con.Configure(ctx, config)
+		is.NoErr(err)
+
+		err = con.Open(ctx) // error is expected, since stream doesn't exist
+		is.True(err != nil)
+
+		// changing the public api just to check for a specific error message
+		// feels a bit of an overkill, so instead we'll just check the error.
+
+		errMsg := err.Error()
+		want := fmt.Sprintf("stream %s does not exist", streamName)
+		if !strings.Contains(errMsg, want) {
+			t.Fatalf("expected error to contain %s, got %s", want, errMsg)
+		}
+	})
+
+	t.Run("errors if option set to false in multi collection mode", func(t *testing.T) {
+		logger := zerolog.New(zerolog.NewTestWriter(t))
+		ctx := logger.WithContext(context.Background())
+		is := is.New(t)
+		con := newDestination()
+
+		config := testutils.GetTestConfig("")
+		config["createIfNotExists"] = "false"
+		err := con.Configure(ctx, config)
+		is.NoErr(err)
+
+		err = con.Open(ctx) // error is expected, since stream doesn't exist
+		is.NoErr(err)
```

Comment:
If it's expected, wouldn't `is.NoErr(err)` fail?

Reply:
Damn just a copy paste mistake, sorry